src/openrct2/object/Object.h

```diff
@@ -111,6 +111,14 @@ typedef struct rct_object_entry_group {
 assert_struct_size(rct_object_entry_group, 8);
 #endif
 
+/* This can be easily casted to an rct_object_entry */
```

Comment:
While the sizes may be the same, it will violate aliasing rules. What's the issue with compiling it as C++ as-is?

Reply:
The 'name' field is 8 `char`s long. Trying to put a string like `"SCGTREES"` into it won't work, as that's 9 `char`s long. Works in C, but not in C++.

Reply:
Then fix it properly: put in 8 chars, not a string.

Reply:
I think it was probably the fact that it wanted to null terminate the string in the initialisers.

Reply:
> Then fix it properly: put in 8 chars, not a string.

`{ 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'I' }` would look horrible.

Reply:
I might have another way.

Reply:
I would probably just turn them to char * lists and ditch the object flags.

Reply:
It needs those object flags later on. My current solution works, in any case, so I'd rather go with that.